templates/Makefile.erb

```diff
@@ -79,8 +79,11 @@ cleanup-components: <%= @components.map {|comp| "#{comp.name}-cleanup" }.join("
 	touch <%= comp.name %>-patch
 
 <%= comp.name %>-configure: <%= comp.name %>-patch <%= list_component_dependencies(comp).join(" ") %>
+	<%- if comp.get_build_dir -%>
+	[ -d <%= comp.get_build_dir %> ] || mkdir <%= comp.get_build_dir %>
```

Comment:
I believe this difference in behavior of `mkdir` and `mkdir -p` is actually part of POSIX. I can try to find the citation if you like.

Aliasing `mkdir` to `mkdir -p` is right up there with aliasing `rm` to `rm -i` - it encourages dumb behavior and fails to teach anyone how the system actually works.